chia/full_node/full_node.py

```diff
@@ -125,7 +125,9 @@ class FullNode:
     _sync_task: Optional[asyncio.Task[None]]
     _transaction_queue: Optional[asyncio.PriorityQueue[Tuple[int, TransactionQueueEntry]]]
     _compact_vdf_sem: Optional[asyncio.Semaphore]
+    _compact_vdf_waiting_sem: Optional[asyncio.Semaphore]
     _new_peak_sem: Optional[asyncio.Semaphore]
+    _new_peak_waiting_sem: Optional[asyncio.Semaphore]
```

Comment:
it seems to me that you could just use plain `int` here, and call them `_new_peak_waiting_counter` and `_compact_vdf_waiting_counter`. I suppose with a `Semaphore` you get the context manager to increment and decrement the counter. But you don't need any of the heavy machinery to manage a queue of waiting tasks and to resume them.

i.e. you don't need the mutex-feature, just the counter-feature.

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/13642